crates/burn-jit/src/codegen/kernel.rs

```diff
@@ -152,11 +369,52 @@ fn execute_settings<'a, R: Runtime, E: JitElement>(
     ExecuteSettings {
         handles_tensors: handles,
         handle_info: info,
-        handle_scalars: scalars,
+        handles_scalars,
         workgroup,
     }
 }
 
+fn create_scalar_handles<R: Runtime, E1: JitElement, E2: JitElement, E3: JitElement>(
+    scalars_0: Option<&[E1]>,
+    scalars_1: Option<&[E2]>,
+    scalars_2: Option<&[E3]>,
+    client: &ComputeClient<R::Server, R::Channel>,
+) -> Vec<Handle<R::Server>> {
+    // It is crucial that scalars follow this order: float, int, uint
+    let scalar_priorities: Vec<usize> = [E1::gpu_elem(), E2::gpu_elem(), E3::gpu_elem()]
+        .iter()
+        .map(|ty: &Elem| match ty {
+            Elem::Float => 0,
+            Elem::Int => 1,
+            Elem::UInt => 2,
+            Elem::Bool => panic!("Bool scalars are not supported"),
+        })
+        .collect();
+
+    let mut handles_scalars = Vec::new();
+    for i in 0..3 {
+        for (j, scalar_priority) in scalar_priorities.iter().enumerate() {
+            if scalar_priority == &i {
```

Comment:
Create an array instead of a vec. We know it's 3 in length, so it's gonna be way more efficient.